### DISCUSSION TITLE
Custom Logic Gates

### DISCUSSION THREAD

**Discussion by Hamguy21** [logisim-evolution/logisim-evolution#1571] [General]
Is it possible to add Custom logic gates?

**Comment by maehne** [comment]
Yes, definitely. The easiest is in form of sub-circuits with optionally customised symbols. A `.circ` file can be loaded into Logisim-evolution using Project -> Load Library -> Logisim-evolution Library. For more performance and flexibility, custom logic gates can be implemented in Java and either integrated into the Logisim-evolution code base (see, e.g., all the recent PRs for different TTL gates) or as a separate JAR library (see the Help -> Guide to Being a Logisim User -> JAR libraries). Note though that that part (as others) of the documentation is known to be outdated. Also JAR libraries have hardly been used by the current maintainers. So, we cannot provide much support in that area. PRs helping in this direction are more than welcome!
